R/validation-emptiness.R

```diff
@@ -16,29 +21,21 @@
 #' isEmpty(df)
 #' isEmpty(df$x[FALSE])
 #' isEmpty(df[FALSE, ])
+#'
+#' # validation helper
+#' validateIsNotEmpty(list(1, 2)) # NULL
+#' # validateIsNotEmpty(NULL) # error
 #' @export
 isEmpty <- function(object) {
   if (is.data.frame(object)) {
     return(nrow(object) == 0)
   }
+
   return(isOfLength(object, 0))
 }
 
-#' Check that the object is not empty
-#'
-#' @inheritParams isEmpty
-#'
-#' @return
-#'
-#' If validations are successful, `NULL` is returned. Otherwise, error is
-#' signaled.
-#'
-#' @examples
-#' # returns `NULL` if of objects are of specified length
-#' validateIsNotEmpty(list(1, 2))
-#'
-#' # error otherwise
-#' # validateIsNotEmpty(NULL)
+
+#' @rdname isEmpty
```

Comment:
@IndrajeetPatil Can you explain why we are changing this (e.g. moving the documentation away from the method?). I know we had a use case like this before and I understood the reasoning but I can't seem to remember why this makes sense. Thanks

Reply:
Because the underlying check and the corresponding validation are inherently linked, and it makes sense that they are documented together. Their code and tests already follow the same pattern, it's only the `.Rd` files that don't.

As a bonus, since they are now documented together, users are immediately exposed to both of them in the same file and understand how the check is utilized for the validation itself.

For example, here is what the new help file looks like for `isEmpty()` function:

<img width="696" alt="Screenshot 2022-04-20 at 21 36 57" src="https://user-images.githubusercontent.com/11330453/164309449-487c2166-93b4-4341-8e2f-24ed5840a30e.png">

Before, if you were already not familiar or cognizant of it, you'd never know the existence of the other function just by looking at one.

Reply:
cool. I knew there was a reason :)